src/box/sql/func.c

```diff
@@ -671,6 +689,7 @@ func_substr_octets(struct sql_context *ctx, int argc, const struct Mem *argv)
 	}
 	char *str = &argv[0].z[start];
 	uint64_t len = MIN(size - start, length);
+	assert(len <= SQL_MAX_LENGTH);
 	ctx->is_aborted = is_str ? mem_copy_str(ctx->pOut, str, len) != 0 :
```

Comment:
Maybe just add this assert to `mem_copy_str()`, `mem_copy_bin()`, and whatever other mem methods require it. It doesn't look right to copy-paste this assert before each usage of mem copy methods.

Reply:
Hi! Thank you for the review! I have added such asserts to all of these functions. Also, I've added an `if` where it's possible for the length to be greater than SQL_MAX_LENGTH, and an assert where it's not possible. Should I drop the last asserts?

Reply:
Yes. The idea is that you only need those asserts somewhere inside mem_copy_str and mem_copy_bin and other mem methods. You don't need to repeat them before every usage of mem methods.